app/Http/Middleware/UserAuthenticated.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace App\Http\Middleware;
+
+use Closure;
+use Illuminate\Http\Request;
+use App\Models\User;
+use Illuminate\Support\Facades\Auth;
+
+class UserAuthenticated
+{
+    /**
+     * Handle an incoming request.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @param  \Closure(\Illuminate\Http\Request): (\Illuminate\Http\Response|\Illuminate\Http\RedirectResponse)  $next
+     * @return \Illuminate\Http\Response|\Illuminate\Http\RedirectResponse
+     */
+    public function handle(Request $request, Closure $next)
+    {
+        if(Auth::check())
+        {
+            $user = Auth::user();
+            if(user->hasRole(1)){
+                return redirect(route('admin.category'));
+            }
+            else if($user->hasRole(0)){
+                return $next($request);
+            }
+        }
+        abort(403);
```

Comment:
I think there is a default middleware that handle if a user is authenticated. So this your middleware can just check the user role.. then you user the auth middleware in your route first, followed by this role middleware. What do you think? Is there any reason why you did it like this?

Reply:
actually, i was thinking about separating the middlewares for admin and user so dat i can easily the middleware for admin on the admin routes and that of the user on the user routes.